File: 1861/Solution.java
class Solution {
    public char[][] rotateTheBox(char[][] box) {
        int ROW = box.length, COL = box[0].length;
        char[][] res = new char[COL][ROW];
        for(char[] a : res) Arrays.fill(a, '.');

        for(int r = 0; r < ROW; r++) {
            int last = COL - 1;
            for(int c = COL - 1; c >= 0; c--) {
                if(box[r][c] == '#'){
                    res[last--][ROW - r - 1] = '#';
                }else if(box[r][c] == '*'){
                    last = c - 1;
                    res[c][ROW - r - 1] = '*';
                }
            }
        }
        return res;
    }
}

File: 1861/Solution.go
func rotateTheBox(box [][]byte) [][]byte {
    R, C := len(box), len(box[0])
    res := make([][]byte, C)
    for i := 0; i < C; i++ {
        res[i] = make([]byte, R)
        for j := 0; j < R; j++ {
            res[i][j] = '.'
        }
    }
    for r := 0; r < R; r++{
        last := C - 1
        for c := C - 1; c >= 0; c-- {
            if box[r][c] == '#' {
                res[last][R - r - 1] = '#'
                last--
            }else if box[r][c] == '*' {
                last = c - 1
                res[c][R - r - 1] = '*'
            }
        }
    }
    return res
}
